src/tutorials/basics/writing-on-the-screen.md

```diff
@@ -0,0 +1,767 @@
+#    app.new_window().event(event).view(view).build().unwrap();  // build a window for the app, handle keystrokes, draw things on it
+#    let mut ui = app.new_ui().build().unwrap(); // create a mutable UI
+#    let message = String::from(
+#        "PRESS 'L' TO LIST ALL FONT IDS\nON THE TERMINAL\n\nPRESS 'C' TO CHANGE FONT\nON SCREEN\n\nPRESS 'ESC' TO QUIT",
+#    ); // text to display
+#
+#    // generate IDs for each widget
+    let widget_ids = Ids {
+        text: ui.generate_widget_id(),
+    };
+#
+#    // load other fonts
+#    let font_map = ui.fonts_mut();  // get the mutable map of fonts for the UI
+#    let font_id = font_map.ids().next().unwrap(); // use default font ID as default
+#    let mut res = font_map.insert_from_file("assets/fonts/retroscape/Retroscape.ttf");
+#    if let Err(e) = res {
+#        println!("{}", e);
+#        panic!("loading font failed");
+#    }
+#    res = font_map.insert_from_file("assets/fonts/voynich/voynich-1.23-webfont.ttf");
+#    if let Err(e) = res {
+#        println!("{}", e);
+#        panic!("loading font failed");
+#    }
+#
+#   // generate the actual model (of the UI)
+#    Model {
+#        ui,
+#        message,
+#        font_id,
+#        widget_ids,
+#    }
+# }
+# // handle events and update the (world) model accordingly
+# fn event(_app: &App, model: &mut Model, event: WindowEvent) {}
+#
+# // handle update event to update the (world) model accordingly
+# fn update(_app: &App, model: &mut Model, _update: nannou::event::Update) {}
+#
+# // display the state of the world
+# fn view(app: &App, model: &Model, frame: Frame) -> Frame {
+#    frame
+# }
+```
+we make an `Ids` struct and generate a widget ID for the text field. `ui.generate_widget_id()`
+is necessary to gurantee unique IDs per widget so each can interact without internal confusion.
+
+Now that the UI widgets are taken care of the can handle the fonts to display the message text.
+We learned already that Nannou tracks fonts in a (global) font map. We get access to it by
+calling `ui.fonts_mut()`. We retrieve all `ids()` from it (in form of an [iterator](https://en.wikipedia.org/wiki/Iterator))
+and get the `next()` ID. Since we are at the beginning this is going to be the ID of the default
+font. But wait? We know that the font is a file ... where is that loaded from? You are right
+in that fonts are files. They come in different formats (like [TrueType](https://en.wikipedia.org/wiki/TrueType),
+or [OpenType](https://en.wikipedia.org/wiki/OpenType)). These fonts are assets for the app.
+Assets are stored in the `assets/` directory. The default for is [Noto (sans regular)](https://www.google.com/get/noto/)
+and it needs to be placed in the `DEFAULT_FONT_PATH`, i.e. "fonts/NotoSans/NotoSans-Regular.ttf".
+We also want different fonts / typefaces. For this tutorial the fonts [retroscape](https://fontlibrary.org/en/font/retroscape)
+and [voynich](https://fontlibrary.org/en/font/voynich-font) are also used. They need to be
```

Comment:
Neither of these URLs are working for me at the moment. Maybe the site is down? Can you confirm that the license for these fonts is permissive enough for us to use them in this tutorial?

Reply:
Maybe. They were working at the time of writing and can still be found in the caches of search engines. Retroscape is Creative Commons CC0, voynich was permissive. I will replace them since the website is down.